### PR TITLE
8328553: Get rid of JApplet in test/jdk/sanity/client/lib/SwingSet2/src/DemoModule.java

### PR DESCRIPTION
**Pull request by lawrence-andrew**
1) Replaced JApplet with JPanel 
2) Tests passed before and after fix

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8328553](https://bugs.openjdk.org/browse/JDK-8328553): Get rid of JApplet in test/jdk/sanity/client/lib/SwingSet2/src/DemoModule.java (**Bug** - P3)

### Reviewers
 * [Harshitha Onkar](https://openjdk.org/census#honkar) (@honkar-jdk - **Reviewer**)
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20450/head:pull/20450` \
`$ git checkout pull/20450`

Update a local copy of the PR: \
`$ git checkout pull/20450` \
`$ git pull https://git.openjdk.org/jdk.git pull/20450/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20450`

View PR using the GUI difftool: \
`$ git pr show -t 20450`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20450.diff">https://git.openjdk.org/jdk/pull/20450.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20450#issuecomment-2267091350)

### CODE REVIEW

**Comment by honkar-jdk** [comment]
Changes look good.

Not related to the fix, but when you click on File > About, following exception is seen due to missing image (resources/images/About.jpg). Maybe this option is not required for the sanity test in which case it can be disabled.

image

About.jpg is available jfc/SwingSet2/ demo version.

```
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException: Cannot invoke "java.net.URL.toExternalForm()" because "location" is null
        at java.desktop/javax.swing.ImageIcon.<init>(ImageIcon.java:232)
        at SwingSet2.createImageIcon(SwingSet2.java:783)
        at SwingSet2$AboutPanel.<init>(SwingSet2.java:1161)
```

**Comment by lawrence-andrew** [comment]
I can file a new ticket and work on the NPE.

File: test/jdk/sanity/client/lib/SwingSet2/src/DemoModule.java [hunk 1]
```diff
@@ -42,10 +42,8 @@
 
 /**
  * A generic SwingSet2 demo module
- *
- * @author Jeff Dinkins
  */
-public class DemoModule extends JApplet {
+public class DemoModule extends JPanel {
```
Comment (prrace) [inline test/jdk/sanity/client/lib/SwingSet2/src/DemoModule.java:46]: But you still import JApplet ! 
See line 34 above.
> import javax.swing.JApplet;

**Comment by honkar-jdk** [comment]
> I can file a new ticket and work on the NPE.
Ok.

**Review by honkar-jdk — APPROVED** [review]
LGTM

**Review by prrace — APPROVED** [review]

**Pull request closed without merging**